### ISSUE TITLE
Solr query function mis-casts date times as dates

### ISSUE THREAD

**Issue by amoeba** [DataONEorg/rdataone#174]
For an XML-formatted Solr result like this:

```{xml}
<result name="response" numFound="1" start="0">
    <doc>
      <str name="identifier">urn:uuid:5e4a483a-8a0e-4993-9f51-8258fb4a0228</str>
      <date name="dateUploaded">2017-03-07T03:25:35.272Z</date>
    </doc>
  </result>
```

The result you get when you run `dataone::query` with `as = 'data.frame' or `as = 'list'` shows the dateUploaded field as a date, rater than a datetime. e.g.,

```
> response <- dataone::query(node, list(q = paste0("identifier:(",
+                                                    paste0(paste0("\"", resource_map_pids, "\"", collapse =" OR "),
+                                                           ")")),
+                                         fl = "identifier,dateUploaded"),
+                              as = "data.frame")
> response
                                     identifier dateUploaded
1 urn:uuid:5e4a483a-8a0e-4993-9f51-8258fb4a0228   2017-03-07
```

I haven't looked at whether this affects other Solr fields at all.

**Comment by gothub** [comment]
@amoeba Good catch! All the <date> fields from the solr result are affected, e.g. 'dateUploaded', 'dateModified', etc. 

The result is now cast to an R `POSIXct` value.

**Comment by amoeba** [comment]
great, thanks a bunch for fixing this up!

**Comment by gothub** [comment]
Fixed in commit 8bc27c8a203633418765ad54e47f05a1e6b6616e
